src/transformers/models/videomae/modeling_tf_videomae.py

```diff
@@ -0,0 +1,1168 @@
+                TFVideoMAELayer(decoder_config, name=f"decoder_layers.{j}")
+                for j in range(config.decoder_num_hidden_layers)
+            ]
+        ]
+
+        self.norm = tf.keras.layers.LayerNormalization(epsilon=config.layer_norm_eps, name="norm")
+        self.head = tf.keras.layers.Dense(decoder_num_labels, name="head") if decoder_num_labels > 0 else tf.identity
+
+    def call(
+        self,
+        hidden_states,
+        return_token_num,
+        output_attentions=False,
+        output_hidden_states=False,
+        return_dict=True,
+    ):
+        # apply Transformer layers (blocks)
+        all_hidden_states = () if output_hidden_states else None
+        all_self_attentions = () if output_attentions else None
+        for i, layer_module in enumerate(self.decoder_layers):
+            if output_hidden_states:
+                all_hidden_states = all_hidden_states + (hidden_states,)
+
+            layer_outputs = layer_module(hidden_states, head_mask=None, output_attentions=output_attentions)
+
+            hidden_states = layer_outputs[0]
+
+            if output_attentions:
+                all_self_attentions = all_self_attentions + (layer_outputs[1],)
+
+        if output_hidden_states:
+            all_hidden_states = all_hidden_states + (hidden_states,)
+
+        if return_token_num > 0:
+            hidden_states = hidden_states[:, -return_token_num:]
+
+        # predictor projection
+        hidden_states = self.norm(hidden_states)
+        logits = self.head(hidden_states)
+
+        if not return_dict:
+            return tuple(v for v in [logits, all_hidden_states, all_self_attentions] if v is not None)
+        return TFVideoMAEDecoderOutput(logits=logits, hidden_states=all_hidden_states, attentions=all_self_attentions)
+
+
+@add_start_docstrings(
+    "The VideoMAE Model transformer with the decoder on top for self-supervised pre-training.",
+    VIDEOMAE_START_DOCSTRING,
+)
+class TFVideoMAEForPreTraining(TFVideoMAEPreTrainedModel):
+    def __init__(self, config):
+        super().__init__(config)
+        self.config = config
+
+        self.videomae = TFVideoMAEMainLayer(config, name="videomae")
+
+        self.encoder_to_decoder = tf.keras.layers.Dense(
+            config.decoder_hidden_size, use_bias=False, name="encoder_to_decoder"
+        )
+        self.position_embeddings = get_sinusoid_encoding_table(
+            self.videomae.embeddings.num_patches, config.decoder_hidden_size
+        )
+
+        self.decoder = TFVideoMAEDecoder(config, name="decoder")
+
+        self.mean = tf.convert_to_tensor(IMAGENET_DEFAULT_MEAN)[None, None, :, None, None]
+        self.std = tf.convert_to_tensor(IMAGENET_DEFAULT_STD)[None, None, :, None, None]
+
+    def build(self, input_shape: tf.TensorShape):
+        self.mask_token = self.add_weight(
+            shape=(1, 1, self.config.decoder_hidden_size), initializer="zeros", name="mask_token"
+        )
+        super().build(input_shape)
```

Comment:
@gante @Rocketknight1 

The PT variant of this model initializes a parameter inside `PretrainedModel`:

https://github.com/huggingface/transformers/blob/8637316e5e94ba0a2493e5df7846f2f23f46eaef/src/transformers/models/videomae/modeling_videomae.py#L742

I followed it in the TF variant and it led to the obvious error:

```
ValueError: You cannot build your model by calling `build` if your layers do not support float type inputs. Instead, in order to instantiate and build your model, call your model on real tensor data (of the correct type).
```

Where do you think we should initialize it (`mask_token`) in the TF variant?